.github/workflows/check-release-branch.yml

```
@@ -38,3 +38,55 @@ jobs:
                     echo "Base branch has commits since merge base"
                     exit 1
                   fi
+
+    check_pr_branch:
+        name: Check PR branch
+        runs-on: ubuntu-latest
+
+        steps:
+            - name: Checkout repository
+              uses: actions/checkout@v4
+
+            - name: Get branch name of the PR
+              id: pr_branch_name
+              run: |
+                  pr_branch_name=$(git branch --show-current)
+                  echo "result=$PR_BRANCH_NAME" >> $GITHUB_OUTPUT
+
+            - name: Assert pull request is created from a RC branch
+              env:
+                  PR_BRANCH_NAME: ${{ steps.pr_branch_name.outputs.result }}
+              run: |
+                  set -x
+                  if [ $PR_BRANCH_NAME != rc/* ]; then
+                      echo "Pull request to release branch must come from RC branch"
+                      exit 1
+                  fi
+
+            - name: Parse version
+              id: version
+              uses: ./.github/parse-version
+
+            - name: Assert release version is not pre-release
+              env:
+                  RELEASE_VERSION: ${{ steps.version.outputs.releaseVersion }}
+              run: |
+                  set -x
+                  echo $RELEASE_VERSION
+                  if [ $RELEASE_VERSION = *-pre ]; then
+                    echo "Release version cannot be pre-release"
+                    exit 1
+                  fi
```

This works, and briefer isn't necessarily better, but this would also work I think:

```suggestion
              if: endsWith(steps.version.outputs.releaseVersion
              run: exit 1
```

---

yeah a matter of portability vs readability. think for this readability is probably preferable so applying this suggestion seems reasonable

---

addressed via https://github.com/digidem/release-automations-expo/pull/7/commits/2ad18326904e827e6c8792bb8385904934aeca1b